boa/src/builtins/function/tests.rs

```diff
@@ -15,11 +13,36 @@ fn check_arguments_object() {
         "#;
 
     eprintln!("{}", forward(&mut engine, init));
-    let expected_return_val = 100;
+
     let return_val = forward_val(&mut engine, "val").expect("value expected");
     assert_eq!(return_val.is_integer(), true);
     assert_eq!(
-        from_value::<i32>(return_val).expect("Could not convert value to i32"),
-        expected_return_val
+        return_val
+            .to_i32(&mut engine)
+            .expect("Could not convert value to i32"),
+        100
+    );
+}
+
+#[test]
+fn check_self_mutating_func() {
+    let realm = Realm::create();
+    let mut engine = Interpreter::new(realm);
+    let func = r#"
+        function x() {
+	        x.y = 3;
+        }
+        var a = x();
+        "#;
+
+    eprintln!("{}", forward(&mut engine, func));
+
+    let return_val = forward_val(&mut engine, "a").expect("value expected");
+    assert_eq!(return_val.is_integer(), true);
+    assert_eq!(
+        return_val
+            .to_i32(&mut engine)
+            .expect("Could not convert value to i32"),
+        3
     );
 }
```

Comment:
```suggestion
#[test]
fn self_mutating_function() {
    let realm = Realm::create();
    let mut engine = Interpreter::new(realm);
    let func = r#"
        function x() {
	        x.y = 3;
        }
        x();
        "#;

    eprintln!("{}", forward(&mut engine, func));

    let y = forward_val(&mut engine, "x.y").expect("value expected");
    assert_eq!(y.is_integer(), true);
    assert_eq!(
        y.to_i32(&mut engine)
            .expect("Could not convert value to i32"),
        3
    );
}
```
This seems like we are testing the return value of the function and not the function object itself